ocs_ci/deployment/acm.py

```diff
@@ -0,0 +1,204 @@
+                logger.exception(
+                    "Failed to download the downloader script from submariner site"
+                )
+                raise
+            tempf = tempfile.NamedTemporaryFile(
+                dir=".", mode="wb", prefix="submariner_downloader_", delete=False
+            )
+            tempf.write(resp.content)
+
+            # Actual submariner binary download
+            cmd = f"bash {tempf.name}"
+            try:
+                run_cmd(cmd)
+            except CommandFailed:
+                logger.exception("Failed to download submariner binary")
+                raise
+
+            # Copy submariner from ~/.local/bin to ocs-ci/bin
+            # ~/.local/bin is the default path selected by submariner script
+            shutil.copyfile(
+                os.path.expanduser("~/.local/bin/subctl"),
+                os.path.join(config.RUN["bin_dir"], "subctl"),
+            )
+
+    def submariner_configure_upstream(self):
+        """
+        Deploy and Configure upstream submariner
+
+        Raises:
+            DRPrimaryNotFoundException: If there is no designated primary cluster found
+
+        """
+        if self.designated_broker_cluster_index < 0:
+            raise DRPrimaryNotFoundException("Designated primary cluster not found")
+
+        # Deploy broker on designated cluster
+        # follow this config switch statement carefully to be mindful
+        # about the context with which we are performing the operations
+        config.switch_ctx(self.designated_broker_cluster_index)
+        logger.info(f"Switched context: {config.cluster_ctx.ENV_DATA['cluster_name']}")
+
+        deploy_broker_cmd = "deploy-broker"
+        try:
+            # run_subctl_cmd(deploy_broker_cmd)
+            pass
+        except CommandFailed:
+            logger.exception("Failed to deploy submariner broker")
+            raise
+
+        # Label the gateway nodes on all non acm cluster
+        # restore_index = config.cur_index
+        # for cluster in get_non_acm_cluster_config():
+        #     config.switch_ctx(cluster.MULTICLUSTER["multicluster_index"])
+        #     gateway_node = self.get_default_gateway_node()
+        #     label_nodes([gateway_node], constants.SUBMARINER_GATEWAY_NODE_LABEL)
+        # config.switch_ctx(restore_index)
+
+        # Join all the clusters (except ACM cluster in case of hub deployment)
+        for cluster in config.clusters:
+            print(len(config.clusters))
+            cluster_index = cluster.MULTICLUSTER["multicluster_index"]
+            if cluster_index != config.get_acm_index():
+                join_cmd = (
+                    f"join --kubeconfig {cluster.RUN['kubeconfig']} "
+                    f"{config.ENV_DATA['submariner_info_file']} "
+                    f"--clusterid c{self.cluster_seq} --natt=false"
+                )
+                try:
+                    # run_subctl_cmd(
+                    #     join_cmd,
+                    # )
```

Comment:
looks like code is commented

Reply:
@prsurve , yep, sorry, I was testing something incrementally so I had it commented and I wanted another branch for Jim's testing so pushed this as is. Soon I will uncomment this.

Reply:
Uncommented the proper lines